app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java

```diff
@@ -88,10 +89,20 @@ public void removeContact(SMSPeer peer) {
      * @return true if peer is present in the database, false otherwise
      */
     public boolean containsPeer(SMSPeer peer) {
-        List<SMSContact> contactList = getAllContacts();
-        for (SMSContact contact : contactList)
-            if (contact.getAddress().equals(peer.getAddress()))
-                return true;
-        return false;
+        return getContactForPeer(peer) != null;
+    }
+
+    /**
+     * Returns the Contact corresponding to a Peer.
+     *
+     * @param peer {@link SMSPeer} to find
+     * @return The Contact with the given Peer address, {@code null} if it does not exist.
+     */
+    @Nullable
+    public SMSContact getContactForPeer(SMSPeer peer) {
+        List<SMSContact> queryResult =
+                contactDatabase.access().getContactsForAddresses(peer.getAddress());
+        if (queryResult == null || queryResult.isEmpty()) return null;
+        return queryResult.get(0);
```

Comment:
It's not possibile to insert more contacts with the same number, is it?

Reply:
Yeah, it's not possible, the get(0) is due to the fact I set up the query to work on a vararg for flexibility, so it might return a List when provided with an array of addresses, here it won't because the number is the primary key and we're querying on a single number